RHDEVDOCS-3750+3704 Tekton Hub with OpenShift Pipelines (#43307)
openshift/openshift-docs: modules/op-installing-and-deploying-tekton-hub-on-an-openshift-cluster.adoc

// This module is included in the following assembly:
//
// *cicd/pipelines/using-tekton-hub-with-openshift-pipelines.adoc

:_content-type: PROCEDURE
[id="op-installing-and-deploying-tekton-hub-on-an-openshift-cluster_{context}"]

> = Installing and deploying {tekton-hub} on an OpenShift cluster

Reviewer: If you refer to the OpenShift product, it helps if you use the {product-title} attribute instead of "OpenShift." This way, our upstream docs can be vendor-neutral and not Red Hat-branded.

[role="_abstract"]
{tekton-hub} is an optional component; cluster administrators cannot install it using the `TektonConfig` custom resource (CR). To install and manage {tekton-hub}, use the `TektonHub` CR.

[NOTE]
====

> If you are using Github Enterprise or Gitlab Enterprise, install and deploy {tekton-hub} in the same network where the enterprise server is running. For example, if enterprise server is running behind VPN, then the Openshift cluster on which you are deploying {tekton-hub} must also be behind VPN.

Reviewer: ```suggestion
If you are using Github Enterprise or Gitlab Enterprise, install and deploy {tekton-hub} in the same network where the enterprise server is running. For example, if the enterprise server is running behind a VPN, deploy {tekton-hub} on a Openshift cluster that is behind the VPN.
```

Reviewer: If you refer to the OpenShift product, use the {product-title} attribute.

Reviewer: This line caught my eye because of the missing articles in "if the enterprise server is running behind a VPN,". I've suggested some further simplifications of the last phrase.

Reviewer: s/Gitlab/GitLab/

====

[discrete]
.Prerequisites
* Ensure that the {pipelines-title} Operator is installed in the default `openshift-pipelines` namespace on the cluster.

[discrete]
.Procedure

. Create a fork of the link:https://github.com/tektoncd/hub[Tekton Hub] repository.

. Clone the forked repository. 

. Update the `config.yaml` file to include at least one user with the following scopes:

> * A user with `agent:create` scope; such a user can set up a cron job that refreshes the {tekton-hub} database after an interval, if there are any changes in the catalog.

Reviewer: ```suggestion
* A user with `agent:create` scope. Such a user can set up a cron job that refreshes the {tekton-hub} database after an interval, if there are any changes in the catalog.
```
or
```suggestion
* A user with `agent:create` scope who can set up a cron job that refreshes the {tekton-hub} database after an interval, if there are any changes in the catalog.
```

Reviewer: Similar suggestions for the next two lines.

* A user with the `catalog:refresh` scope; such a user can refresh the catalog and all resources in the database of the {tekton-hub}.
* A user with the `config:refresh` scope; such a user can get additional scopes. 
+
[source,yaml]
----
...
scopes:
- name: agent:create
  users: <username_registered_with_the_Git_repository_hosting_service_provider>
- name: catalog:refresh
  users: <username_registered_with_the_Git_repository_hosting_service_provider>
- name: config:refresh
  users: <username_registered_with_the_Git_repository_hosting_service_provider>
...
----
+
The supported service providers are GitHub, GitLab, and BitBucket.

. Create an OAuth application with your Git repository hosting provider, and note the Client ID and Client Secret. 
* For a GitHub OAuth application, set the `Homepage URL` and the `Authorization callback URL` as `<auth-route>`.
* For a GitLab OAuth application, set the `REDIRECT_URI` as `<auth-route>/auth/gitlab/callback`.
* For a BitBucket OAuth application, set the `Callback URL` as `<auth-route>`.

> . Edit the following fields in the `<project_root>/config/02-api/20-api-secret.yaml` file for the {tekton-hub} API secret:

Reviewer 2: Where is this file when admin has installed Hub through the operator?

Author: As per my understanding, if the admin has forked and cloned the Tekton Hub repository as instructed in the previous steps, then the file should be available in the mentioned location even when Hub is installed using the Operator. 

The documentation reflects the relative path as same as that of https://github.com/tektoncd/hub/blob/main/config/02-api/20-api-secret.yaml.
